flang/lib/Semantics/resolve-directives.cpp

```diff
@@ -1454,6 +1464,12 @@ void OmpAttributeVisitor::ResolveOmpName(
         AddToContextObjectWithDSA(*resolvedSymbol, ompFlag);
       }
     }
+  } else if (ompFlag == Symbol::Flag::OmpCriticalLock) {
+    // Create a new symbol.
+    const auto pair{GetContext().scope.try_emplace(
+        name.source, Attrs{}, UnknownDetails{})};
```

Comment:
That patch has been lying idle for a while since @Sameeranjoshi moved to another team. But I hear that he is back. So there is a good chance that this will land soon with some changes (including this change). Otherwise, I will take the patch forward.